### PR TITLE
nighthawk package init

### PR DESCRIPTION
**Pull request by kumarabd**
Signed-off-by: kumarabd <abishekkumar92@gmail.com>

**Description**

Protobuf and library for nighthawk service

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: pkg/client/transform.go [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+		Results: results,
+	}
+	input, err := utils.Marshal(t)
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println("input: ", input)
+
+	command := "/private/var/tmp/_bazel_abishekk/eaf1167d72f4772496616c435b301da8/execroot/nighthawk/bazel-out/darwin-opt/bin/nighthawk_output_transform"
+	cmd := exec.Command(command, "--output-format", "fortio")
```
Comment (leecalcote) [inline pkg/client/transform.go:247]: Always fortio. I think this is ok.

File: pkg/client/error.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	"github.com/layer5io/meshkit/errors"
+)
+
+const (
+	ErrGRPCDialCode        = "1000"
+	ErrInvalidEndpointCode = "1001"
+)
+
+var (
+	ErrInvalidEndpoint = errors.NewDefault(ErrInvalidEndpointCode, "Endpoint not reachable")
```
Comment (leecalcote) [inline pkg/client/error.go:13]: Probably cause and suggested remediation?

File: .github/workflows/build-and-publish.yaml [hunk 3]
```diff
@@ -33,7 +33,7 @@ jobs:
     strategy:
       max-parallel: 10
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macos-latest]
```
Comment (leecalcote) [inline .github/workflows/build-and-publish.yaml:36]:
```suggestion
        os: [ubuntu-latest]
```

**Review by leecalcote — CHANGES_REQUESTED** [review]
A few lint checks to tend to and two error codes to enhance.

**Comment by leecalcote** [comment]
@kumarabd good to merge after errors are updated.

File: .github/workflows/build-and-publish.yaml (hunk 3 above)
Comment (kumarabd) [reply .github/workflows/build-and-publish.yaml]: @leecalcote so no Darwin builds for nighthawk?

File: pkg/client/error.go (hunk 2 above)
Comment (leecalcote) [reply pkg/client/error.go:14]: Probable cause: "Endpoint is unavailable or endpoint is unreachable" - sound right?

File: .github/workflows/build-and-publish.yaml (hunk 3 above)
Comment (leecalcote) [reply .github/workflows/build-and-publish.yaml]: @kumarabd, How much time do they take to build? Will this org run out of workflow time?

File: pkg/client/error.go [hunk 4]
```diff
@@ -11,7 +11,7 @@ const (
 )
 
 var (
-	ErrInvalidEndpoint = errors.NewDefault(ErrInvalidEndpointCode, "Endpoint not reachable")
```
Comment (leecalcote) [inline pkg/client/error.go]: @kumarabd probable cause and suggested remediation are missing here.

**Pull request merged**